### ISSUE TITLE
"Payload could not be written as JSON" error when using call mediator in blocking mode

### ISSUE THREAD

**Issue by sashikamw** [wso2/product-ei#1805]
**Description:**
Error when call mediator used in blocking mode

**Affected Product Version:**
EI 6.1.1 (wso2ei-6.1.1.1512875284190)

**OS, DB, other environment details and versions:**    
OS - ubuntu 15.10
DB - H2

**Steps to reproduce:**
1. create below API
```
<api xmlns="http://ws.apache.org/ns/synapse" name="Backend4" context="/userinfomation">
   <resource methods="POST">
      <inSequence>
         <call blocking="true">
            <endpoint>
               <http uri-template="http://www.mocky.io/v2/5a6851eb2e00002509d5b303/userinfo"/>
            </endpoint>
         </call>
         <respond/>
      </inSequence>
   </resource>
</api>
                        
```

Observations
```
[2018-01-25 17:26:00,529] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "POST /userinfomation HTTP/1.1[\r][\n]"
[2018-01-25 17:26:00,529] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "Host: 10.100.8.9:8280[\r][\n]"
[2018-01-25 17:26:00,530] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "User-Agent: curl/7.43.0[\r][\n]"
[2018-01-25 17:26:00,530] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "Accept: */*[\r][\n]"
[2018-01-25 17:26:00,530] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "Content-Type: application/json[\r][\n]"
[2018-01-25 17:26:00,530] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "Content-Length: 11[\r][\n]"
[2018-01-25 17:26:00,530] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "[\r][\n]"
[2018-01-25 17:26:00,530] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 >> "{"a" : "b"}"
[2018-01-25 17:26:00,981] [EI-Core] ERROR - JsonUtil #writeAsJson. Payload could not be written as JSON. MessageID: urn:uuid:3f796012-2551-4a9c-a034-0d1365b3d60c
[2018-01-25 17:26:01,128] [EI-Core] ERROR - JsonUtil #writeAsJson. Payload could not be written as JSON. MessageID: urn:uuid:3f796012-2551-4a9c-a034-0d1365b3d60c
[2018-01-25 17:26:01,282] [EI-Core] ERROR - JsonUtil #writeAsJson. Payload could not be written as JSON. MessageID: urn:uuid:3f796012-2551-4a9c-a034-0d1365b3d60c
[2018-01-25 17:26:01,433] [EI-Core] ERROR - JsonUtil #writeAsJson. Payload could not be written as JSON. MessageID: urn:uuid:3f796012-2551-4a9c-a034-0d1365b3d60c
[2018-01-25 17:26:01,435] [EI-Core]  INFO - HTTPSender Unable to sendViaPost to url[http://www.mocky.io/v2/5a6851eb2e00002509d5b303/userinfo]
org.apache.axis2.AxisFault: Payload could not be written as JSON.
	at org.apache.synapse.commons.json.JsonUtil.writeAsJson(JsonUtil.java:289)
	at org.apache.synapse.commons.json.JsonStreamFormatter.writeTo(JsonStreamFormatter.java:70)
	at org.apache.axis2.transport.http.AxisRequestEntity.writeRequest(AxisRequestEntity.java:84)
	at org.apache.commons.httpclient.methods.EntityEnclosingMethod.writeRequestBody(EntityEnclosingMethod.java:499)
	at org.apache.commons.httpclient.HttpMethodBase.writeRequest(HttpMethodBase.java:2114)
	at org.apache.commons.httpclient.HttpMethodBase.execute(HttpMethodBase.java:1096)
	at org.apache.commons.httpclient.HttpMethodDirector.executeWithRetry(HttpMethodDirector.java:398)
	at org.apache.commons.httpclient.HttpMethodDirector.executeMethod(HttpMethodDirector.java:171)
	at org.apache.commons.httpclient.HttpClient.executeMethod(HttpClient.java:397)
	at org.apache.axis2.transport.http.AbstractHTTPSender.executeMethod(AbstractHTTPSender.java:682)
	at org.apache.axis2.transport.http.HTTPSender.sendViaPost(HTTPSender.java:195)
	at org.apache.axis2.transport.http.HTTPSender.send(HTTPSender.java:77)
	at org.apache.axis2.transport.http.CommonsHTTPTransportSender.writeMessageWithCommons(CommonsHTTPTransportSender.java:451)
	at org.apache.axis2.transport.http.CommonsHTTPTransportSender.invoke(CommonsHTTPTransportSender.java:278)
	at org.apache.axis2.engine.AxisEngine.send(AxisEngine.java:442)
	at org.apache.axis2.description.OutInAxisOperationClient.send(OutInAxisOperation.java:430)
	at org.apache.axis2.description.OutInAxisOperationClient.executeImpl(OutInAxisOperation.java:225)
	at org.apache.axis2.client.OperationClient.execute(OperationClient.java:149)
	at org.apache.synapse.message.senders.blocking.BlockingMsgSender.sendReceive(BlockingMsgSender.java:302)
	at org.apache.synapse.message.senders.blocking.BlockingMsgSender.send(BlockingMsgSender.java:211)
	at org.apache.synapse.mediators.builtin.CallMediator.handleBlockingCall(CallMediator.java:150)
	at org.apache.synapse.mediators.builtin.CallMediator.mediate(CallMediator.java:113)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:97)
	at org.apache.synapse.mediators.AbstractListMediator.mediate(AbstractListMediator.java:59)
	at org.apache.synapse.mediators.base.SequenceMediator.mediate(SequenceMediator.java:158)
	at org.apache.synapse.rest.Resource.process(Resource.java:343)
	at org.apache.synapse.rest.API.process(API.java:399)
	at org.apache.synapse.rest.RESTRequestHandler.apiProcess(RESTRequestHandler.java:123)
	at org.apache.synapse.rest.RESTRequestHandler.dispatchToAPI(RESTRequestHandler.java:101)
	at org.apache.synapse.rest.RESTRequestHandler.process(RESTRequestHandler.java:69)
	at org.apache.synapse.core.axis2.Axis2SynapseEnvironment.injectMessage(Axis2SynapseEnvironment.java:304)
	at org.apache.synapse.core.axis2.SynapseMessageReceiver.receive(SynapseMessageReceiver.java:78)
	at org.apache.axis2.engine.AxisEngine.receive(AxisEngine.java:180)
	at org.apache.synapse.transport.passthru.ServerWorker.processNonEntityEnclosingRESTHandler(ServerWorker.java:326)
	at org.apache.synapse.transport.passthru.ServerWorker.processEntityEnclosingRequest(ServerWorker.java:372)
	at org.apache.synapse.transport.passthru.ServerWorker.run(ServerWorker.java:151)
	at org.apache.axis2.transport.base.threads.NativeWorkerPool$1.run(NativeWorkerPool.java:172)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
[2018-01-25 17:26:01,460] [EI-Core]  INFO - LogMediator To: http://www.mocky.io/v2/5a6851eb2e00002509d5b303/userinfo, MessageID: urn:uuid:1de7fc5b-8982-4507-bc9b-1f6aefb6f1e7, Direction: request, MESSAGE = Executing default 'fault' sequence, ERROR_CODE = 401001, ERROR_MESSAGE = Payload could not be written as JSON., Payload: {"a" : "b"}
[2018-01-25 17:26:01,462] [EI-Core]  WARN - PassThroughTransportUtils Unable to set the HTTP status code from the property HTTP_SC with value: 
[2018-01-25 17:26:01,467] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 << "HTTP/1.1 202 Accepted[\r][\n]"
[2018-01-25 17:26:01,467] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 << "Date: Thu, 25 Jan 2018 11:56:01 GMT[\r][\n]"
[2018-01-25 17:26:01,468] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 << "Transfer-Encoding: chunked[\r][\n]"
[2018-01-25 17:26:01,468] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 << "[\r][\n]"
[2018-01-25 17:26:01,468] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 << "0[\r][\n]"
[2018-01-25 17:26:01,468] [EI-Core] DEBUG - wire HTTP-Listener I/O dispatcher-1 << "[\r][\n]"

```
Note - If we add a log full this issue will not get reproduced

**Comment by sashikamw** [comment]
Changing the priority to highest as this is a basic use case

**Comment by riyafa** [comment]
Please refer mail on dev mailing list: http://wso2.markmail.org/message/oos6kvrflz4tobgb?q=list:org%2Ewso2%2Edev

**Comment by Vathsan** [comment]
Fixed in [#1301](https://github.com/wso2/wso2-synapse/pull/1301)
